word_class.rb

```diff
@@ -0,0 +1,215 @@
+require "random-word"
+require "colorize"
+require_relative 'cactus.rb'
+
+class WordClass
+  attr_reader :word, :gameboard, :used_guesses, :guess_count
+
+  def initialize
+    @word = ""
+    @gameboard = ""
+    @guess_count = 0
+    @used_guesses = []
+    @guess_ind = []
+    @current_guess =""
+  end
+
+  #generate_word uses RandomWord gem to create a random english word of between
+  #4 and 8 characters.  It excludes words that have "-" and "_" as characters.
+  def generate_word
+    @word = RandomWord.nouns(not_shorter_than: 4, not_longer_than: 8).next
+    while @word.include?("_") || word.include?("-")
+      @word = RandomWord.nouns(not_shorter_than: 4, not_longer_than: 8).next
+    end
+    return @word
+  end
+
+  def generate_gameboard
+    @gameboard = Array.new(@word.length, "_")
+  end
+
+  #start_game generates the word and the empty gameboard
+  def start_game
+    generate_word
+    generate_gameboard
+  end
+
+  #gets a guess from the user and prompts them in a way appropriate to the
+  #number of guesses they have already made
+  def obtain_guess
+
+    if @used_guesses.length > 1
+      puts "\nYou have made these guesses: #{@used_guesses.join(" ")}".colorize(:white)
+    elsif @used_guesses.length == 1
+      puts "\nYou have already guessed: #{@used_guesses.join(" ")}".colorize(:white)
+    end
+    puts "\nPlease guess a letter".colorize(:white)
+
+    loop do
+      @current_guess = gets.chomp
+      break if single_character? && is_a_letter?(@current_guess) && already_guessed?
+      puts "Woah woah woah. I only accept single letters that have not already been guessed. Try again.".colorize(:red)
```

Comment:
I'm not sure if I agree with making the current guess an instance variable here, though it certainly solves the problem.

My thinking is, the current guess is only important within the `turn` method and methods called by it. It doesn't need to persist beyond a single invocation of `turn`. This means it's not really part of the long-term state of the game, which is what instance variables are intended for.

Instead, you might use a local variable that is returned from `obtain_guess` and passed into any other method that needs it as a parameter.